### ISSUE TITLE
[UserStory] Add clear build and 'pre-requisites' to Contributing.md

### ISSUE THREAD

**Issue by CoffeeTulip** [MetroCS/redistricting#201]
# User Story
### Essential components
- [ ] Title describes the story
- [ ] Stakeholder type is identified
- [ ] Outcome is described
- [ ] Rationale is explicit
- [ ] Acceptance criteria are verifiable and from the perspective of the stakeholder

## Story
**As a**  _contributor_  **I want**  _Pre-requisite and step by step Build instructions_  in the contributing.md file
**so that** anyone who wants to contribute can get set up with no problem. 

## Acceptance Criteria
_(Rules or scenarios are acceptable formats.)_

- [ ] * The contributing.md file clearly states what languages, packages, and etc you should have installed before you do anything
- [ ] * Step by step instructions with terminal/command lines to follow for building 
- [ ] * Links to respective websites for packages 

## Supporting Information
Related to epic #80, the contributing.md file could use a much more organized and chronological set of more specific instructions on what technologies are needed and how to build the project as well.

**Comment by CoffeeTulip** [comment]
I would like to be assigned this issue.

**Comment by jody** [comment]
"Before you do anything..." seems like hyperbole. That also makes it hard to validate.

Questions about the "step by step instructions with terminal/command lines" include: "Which OS and user interface are being described?" and "How many specific contexts need to be addressed in the documentation?"  It seems to me that a basic *nix OS and standard shell (bash, csh, zsh, ...) would be appropriate.  We just need to make some decisions.

What is meant by "packages"?

**Comment by CoffeeTulip** [comment]
After having talked about what we expect our audience to already know, Ive decided it better to suggest adding work-arounds to the contributing.md document for specific machines and problems that we've notice tend to arise when you try to follow the existing document. 
More specifically, Id like to add two small work-arounds for users who try to build with a mac and run into a build error like 'ant isn't a recognized command', and for Ubuntu users who have a problem building the jar files. Two small additions to the document so that future users can build and avoid those build problems.

**Comment by jody** [comment]
I agree. The approach you suggest is both highly valuable and feasible.

**Comment by jody** [comment]
Please update the Acceptance Criteria of this issue to reflect your new intent. (Once we agree upon those, I will add this issue to the project.)

**Comment by CoffeeTulip** [comment]
We agreed on adding the 'workarounds' under the apache ant package header, as more of 'for your convenience' tips for installing ant using package managers for specific machines.

**Comment by CoffeeTulip** [comment]
I'd like to suggest adding Junit as a tool in the contributing file. When you build this file, the build tool(ant) may be unable to locate the Junit library during the build, so users should make sure the JUnit JAR file is present in their project or is available in a directory that Ant can access.

**Comment by jody** [comment]
> I'd like to suggest adding Junit as a tool in the contributing file.

I agree! Contributors would benefit from more extensive information about development tools, especially those that perform quality assessment on contributed code.
